libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionTest.kt

```diff
@@ -354,4 +366,197 @@ class MapboxTripSessionTest {
 
         verify { navigator.setRoute(route) }
     }
+
+    @Test
+    fun unregisterAllLocationObservers() {
+        every { routeProgress.bannerInstructions() } returns null
+        every { routeProgress.voiceInstructions() } returns null
+
+        tripSession.start()
+        val observer: LocationObserver = mockk(relaxUnitFun = true)
+        tripSession.registerLocationObserver(observer)
+        tripSession.unregisterAllLocationObservers()
+
+        locationCallbackSlot.captured.onSuccess(locationEngineResult)
+
+        verify(exactly = 0) { observer.onRawLocationChanged(location) }
+        assertEquals(location, tripSession.getRawLocation())
+
+        tripSession.stop()
+    }
+
+    @Test
+    fun unregisterAllRouteProgressObservers() {
+        val parentJob = SupervisorJob()
+        val testScope = CoroutineScope(parentJob + coroutineRule.testDispatcher)
```

Comment:
Do we actually need this? Couldn't https://github.com/mapbox/mapbox-navigation-android/blob/75f7ef798bea81b23e48858b0d1551c2c093bbd4/libtesting/src/main/java/com/mapbox/navigation/testing/MainCoroutineRule.kt#L16-L17 be used?